### PR TITLE
Upgrade to 7.10.2

### PR DESCRIPTION
**Pull request by eirsep**
Upgrades es dependency to 7.10.2 and version compatibility changes

### CODE REVIEW

**Review by Bukhtawar — COMMENTED** [review]
Thanks Surya let's verify assertions for progress listener

File: build.gradle [hunk 1]
```diff
@@ -1,9 +1,9 @@
 buildscript {
     ext {
-        es_version = '7.9.1'
+        es_version = '7.10.2'
         distribution = 'oss-zip'
         es_group = "org.elasticsearch"
-        opendistroVersion = '1.11.1.0'
+        opendistroVersion = '1.12.0'
```
Comment (Bukhtawar) [inline build.gradle:6]: This needs to be `1.13.0`

File: src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/rest/RestSubmitAsynchronousSearchAction.java [hunk 2]
```diff
@@ -51,7 +51,7 @@
 
     @Override
     public String getName() {
-        return "async_search_action";
+        return "asynchronous_search_action";
```
Comment (Bukhtawar) [inline src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/rest/RestSubmitAsynchronousSearchAction.java:54]: `submit_ asynchronous_search_action `

File: src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/task/AsynchronousSearchTask.java [hunk 3]
```diff
@@ -69,6 +68,7 @@ public String getDescription() {
         Strings.arrayToDelimitedString(request.getSearchRequest().types(), ",", sb);
         sb.append("], ");
         sb.append("search_type[").append(request.getSearchRequest().searchType()).append("], ");
+        sb.append("keep_on_completion[").append(request.getKeepOnCompletion()).append("], ");
```
Comment (Bukhtawar) [inline src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/task/AsynchronousSearchTask.java:71]: Lets add assertion on description

**Review by Bukhtawar — APPROVED** [review]

**Review by anshul291995 — APPROVED** [review]

**Pull request merged**